src/hotspot/share/runtime/thread.hpp

```diff
@@ -737,6 +737,23 @@ class Thread: public ThreadShadow {
   // creation fails due to lack of memory, too many threads etc.
   bool set_as_starting_thread();
 
+private:
+  os::ThreadCrashProtection *_crash_protection;
+
+public:
+  bool has_crash_protection() const { return _crash_protection != NULL; }
+  void set_crash_protection(os::ThreadCrashProtection *crash_protection) {
+    _crash_protection = crash_protection;
+  }
+#ifndef _WINDOWS
+  void check_crash_protection(int sig) const {
+    assert(this == Thread::current(), "should only call check_crash_protection() on self");
+    if (_crash_protection != NULL) {
+      _crash_protection->check_crash_protection(sig);
+    }
+  }
+#endif
+
```

Comment:
Could this be pushed down into osThread ?

Reply:
Yes, it can but it's not that clean I think given the osthread indirections. Here's how it looks: https://github.com/pchilano/jdk/commit/73a41ad867b4b7466cdddc87173828b4e80f8179

I think another alternative could be to remove the "#ifndef _WINDOWS" clause and define an empty check_crash_protection() method in os_windows.hpp